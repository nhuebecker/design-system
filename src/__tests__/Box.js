import React from 'react'
import renderer from 'react-test-renderer'
import { Box, theme } from '..'

describe('Box', () => {
  test('renders', () => {
    const json = renderer.create(<Box />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('width prop sets width', () => {
    const json = renderer.create(<Box width={1 / 2} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('width', '50%')
  })

  test('m prop sets margin', () => {
    const json = renderer.create(<Box m={1} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('margin', theme.space[1] + 'px')
  })

  test('p prop sets padding', () => {
    const json = renderer.create(<Box p={1} />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('padding', theme.space[1] + 'px')
  })

  test('color prop sets color', () => {
    const json = renderer.create(<Box color="warning" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('color', theme.colors.warning)
  })

  test('bg prop sets background color', () => {
    const json = renderer.create(<Box bg="accent" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', theme.colors.accent)
  })

  test('bg prop sets custom background color', () => {
    const json = renderer.create(<Box bg="#ff6d00" />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('background-color', '#ff6d00')
  })
})
