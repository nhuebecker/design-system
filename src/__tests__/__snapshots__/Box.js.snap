// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box bg prop sets background color 1`] = `
.c0 {
  background-color: #7280ed;
}

<div
  className="c0"
/>
`;

exports[`Box bg prop sets custom background color 1`] = `
.c0 {
  background-color: #ff6d00;
}

<div
  className="c0"
/>
`;

exports[`Box color prop sets color 1`] = `
.c0 {
  color: #e4732d;
}

<div
  className="c0"
/>
`;

exports[`Box m prop sets margin 1`] = `
.c0 {
  margin: 6px;
}

<div
  className="c0"
/>
`;

exports[`Box p prop sets padding 1`] = `
.c0 {
  padding: 6px;
}

<div
  className="c0"
/>
`;

exports[`Box renders 1`] = `
<div
  className=""
/>
`;

exports[`Box width prop sets width 1`] = `
.c0 {
  width: 50%;
}

<div
  className="c0"
/>
`;
